src/gei/Commands/MigrateOrgCommand.cs

```diff
@@ -0,0 +1,137 @@
+using System.Threading.Tasks;
+using OctoshiftCLI.Contracts;
+using OctoshiftCLI.Extensions;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class MigrationOrgCommand : Command
+    {
+        private readonly OctoLogger _log;
+        private readonly ITargetGithubApiFactory _targetGithubApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+        private const string DEFAULT_GITHUB_BASE_URL = "https://github.com";
+
+        public MigrationOrgCommand(OctoLogger log, ITargetGithubApiFactory targetGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider) : base("migrate-repo")
+        {
+            _log = log;
+            _targetGithubApiFactory = targetGithubApiFactory;
+            _environmentVariableProvider = environmentVariableProvider;
+
+            Description = "Invokes the GitHub APIs to migrate the org, its teams and the repositories.";
+
+            var githubSourceOrg = new Option<string>("--github-source-org")
+            {
+                IsRequired = true,
+                Description = "Uses GH_SOURCE_PAT env variable or --github-source-pat option. Will fall back to GH_PAT or --github-target-pat if not set."
+            };
+            var githubTargetOrg = new Option<string>("--github-target-org")
+            {
+                IsRequired = true,
+                Description = "Uses GH_PAT env variable or --github-target-pat option."
+            };
+            var ssh = new Option("--ssh")
+            {
+                IsRequired = false,
+                IsHidden = true,
+                Description = "Uses SSH protocol instead of HTTPS to push a Git repository into the target repository on GitHub."
+            };
+            var githubSourcePat = new Option<string>("--github-source-pat")
+            {
+                IsRequired = false
+            };
+            var githubTargetPat = new Option<string>("--github-target-pat")
+            {
+                IsRequired = false
+            };
+            var verbose = new Option("--verbose")
+            {
+                IsRequired = false
+            };
+
+            AddOption(githubSourceOrg);
+            AddOption(githubTargetOrg);
+
+            AddOption(ssh);
+            AddOption(githubSourcePat);
+            AddOption(githubTargetPat);
+            AddOption(verbose);
+
+            Handler = CommandHandler.Create<MigrationOrgCommandArgs>(Invoke);
+        }
+
+        public async Task Invoke(MigrationOrgCommandArgs args)
+        {
+            if (args is null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
+            _log.Verbose = args.Verbose;
+
+            LogAndValidateOptions(args);
+
+            var githubApi = _targetGithubApiFactory.Create(targetPersonalAccessToken: args.GithubTargetPat);
+
+            var githubOrgId = await githubApi.GetOrganizationId(args.GithubTargetOrg);
```

Comment:
We do not need the target org id as it won't exist we need the enterprise id instead.